packages/core/src/App/Containers/Layout/app-contents.jsx

```diff
@@ -1,67 +1,132 @@
 import { isMobile } from '@deriv/shared/utils/screen';
 import { connect } from 'Stores/connect';
-// import InstallPWA    from './install-pwa.jsx';
+import BinarySocket from '_common/base/socket_base';
+import { isEuCountry } from '_common/utility';
+import CookieBanner from '../../Components/Elements/CookieBanner/cookie-banner.jsx';
+import { isBrowser, WindowContext } from '../window-context';
+import { LocationProvider } from '../location-context';
 
-const AppContents = ({
-    // addNotificationBar,
-    children,
-    identifyEvent,
-    is_app_disabled,
-    is_cashier_visible,
-    is_mt5_page,
-    is_positions_drawer_on,
-    is_route_modal_on,
-    pageView,
-    // setPWAPromptEvent,
-}) => {
-    // Segment page view trigger
-    identifyEvent();
-    pageView();
+const has_dataLayer = isBrowser() && window.dataLayer;
+const cookie_expires = 7;
 
-    // if (is_logged_in) {
-    // TODO: uncomment these after the issues with showing the prompt too often and in the app are fixed
-    // window.addEventListener('beforeinstallprompt', e => {
-    //     console.log('Going to show the installation prompt'); // eslint-disable-line no-console
-    //
-    //     e.preventDefault();
-    //
-    //     setPWAPromptEvent(e);
-    //     addNotificationBar({
-    //         content : <InstallPWA />,
-    //         autoShow: 10000, // show after 10 secs
-    //         msg_type: 'pwa',
-    //     });
-    // });
-    // }
-    return (
-        <div
-            id='app_contents'
-            className={classNames('app-contents', {
-                'app-contents--show-positions-drawer': is_positions_drawer_on,
-                'app-contents--is-disabled': is_app_disabled,
-                'app-contents--is-mobile': isMobile(),
-                'app-contents--is-route-modal': is_route_modal_on,
-                'app-contents--is-scrollable': is_mt5_page || is_cashier_visible,
-            })}
-        >
-            <MobileWrapper>{children}</MobileWrapper>
-            <DesktopWrapper>
-                {/* Calculate height of user screen and offset height of header and footer */}
-                <ThemedScrollbars autoHide style={{ height: 'calc(100vh - 83px)' }}>
-                    {children}
-                </ThemedScrollbars>
-            </DesktopWrapper>
-        </div>
-    );
-};
+class AppContents extends React.Component {
+    static contextType = WindowContext;
+    state = {
+        show_cookie_banner: false,
+        clients_country: Cookies.get('clients_country'),
+        is_window_loaded: false,
+    };
+
+    async componentDidMount() {
+        this.props.identifyEvent();
+        this.props.pageView();
+        if (!this.state.clients_country) {
+            const website_status = await BinarySocket.wait('website_status');
+            const res = website_status.website_status.clients_country;
+            this.setState({ clients_country: res });
```

Comment:
Is it possible to use regular class properties here? `state` will force a re-render on every country, while I think it would be better to only re-render when there is "EU" country.